src/main/java/com/flipkart/yak/core/ThreadedCompactionJob.java

```diff
@@ -39,15 +41,38 @@ public void init(CompactionContext context) throws ConfigurationException {
 
     @Override
     public void run() {
-        Thread.currentThread().setName(compactionContext.getTableName()+"-"+compactionContext.getNameSpace());
+        if(compactionSchedule.isPrompt()) {
+            Thread.currentThread().setName(compactionContext.getTableName()+"-"+compactionContext.getNameSpace()+"-"+"Prompt");
+        } else {
+            Thread.currentThread().setName(compactionContext.getTableName()+"-"+compactionContext.getNameSpace());
+        }
         MDC.put("JOB", Thread.currentThread().getName());
         log.info("starting compact-cron for : {}", this.getCompactionContext());
         while(true) {
+
+            /*
+            If prompt job , and its life cycle has already ended , exit
+             */
+            if(compactionSchedule.isPrompt() && ScheduleUtils.hasExpired(compactionSchedule, Instant.now())) {
+                Thread.currentThread().interrupt();
```

Comment:
Why do you need to interrupt it ? 
Can it die its own death naturally?

If it has crosses timestamp, it means lifespan is over. Does it need to interrupt anything ?

Reply:
hmmm earlier took safety measure to kill it once job is done .
But it can die its own death naturally as well , have modified it now.